compiler/src/iree/compiler/Codegen/Common/DecomposeBatchMmt4DOps.cpp

```diff
@@ -59,12 +101,14 @@ struct ConvertBatchMmt4DtoMmt4DPattern
         RankedTensorType::Builder(lhsType).dropDim(0);
     auto reducedLhs = tensor::createCanonicalRankReducingExtractSliceOp(
         rewriter, loc, lhs, reducedLhsType);
+    (void)reduceDefiningOp(rewriter, lhs);
```

Comment:
Can we signal an error if there is a failure?

Reply:
If this fails, we don't want to return an error because we still want to reduce the batch dimension of  the inputs. I had `reduceDefiningOp` return `LogicalResult` so it was clear that it will not always succeed, but there is nothing that needs to happen if it fails. I can return a `notifyMatchFailure` inside of `reduceDefiningOp` for verbosity though.

Reply:
I think returning success in l.37 is more proper. It means that there is no need to rank-reduce defining op.

For the failure from linalg::dropUnitDims, we probably want to signal the error. It looks wrong to me if we hit a failure in linalg::dropUnitDims. In the context, we can capture the failure. What do you think?

Reply:
Ok, that makes sense. Before, I hadn't wanted to return failure when linalg::dropUnitDims failed because it will return failure even if the `options.controlFn` failed to match, but I can just move the matching outside of dropUnitDims.